### PR TITLE
Add extensible logic for finding WRF build, e.g., in ../WRF-4.0.3

### PR DESCRIPTION
**Pull request by mgduda**
This merge replaces the logic in `arch/preamble` for locating compiled WRF source
code with more extensible logic in the `configure` script.

The path to the compiled WRF source can now be specified by the environment
variable `$WRF_DIR`, or an attempt will be made to find the compiled source in
`../WRF-4.0.3`, `../WRF-4.0.2`, ..., `WRFV3`.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -150,8 +150,43 @@ if [ $? -eq 0 ] ; then
...
+      exit
+   fi
+
+   echo "Using WRF I/O library in WRF build identified by \$WRF_DIR: ${WRF_DIR}"
+   wrf_dir=$WRF_DIR
+fi
+
+if [ $wrf_dir == "none" ]; then
+   echo ""
+   echo " No compiled WRF code found... Check that WRF has been compiled or whatever"
```
Comment (kkeene44) [inline configure]: @mgduda 
Did you mean to leave this as "....or whatever?"

Comment (mgduda) [reply configure]: That was a test to see how closely anyone checks these PRs. A+. I'll push an amended commit that improves that message.

Comment (mgduda) [reply configure]: I've just pushed an amended commit. Hopefully the message is more informative now.

**Comment by kkeene44** [comment]
@mgduda 
I accidentally tested out probably every scenario for this and everything seems to work great. I like this concept. I'll approve it.

**Pull request merged**